### PR TITLE
Fix download dialog displaying long filenames

### PR DESCRIPTION
**Pull request by mavduevskiy**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by mavduevskiy** [comment]
Screenies

before_mark_up
before
after
after_mark_up

File: app/src/androidTest/java/org/mozilla/fenix/helpers/TestAssetHelper.kt [hunk 1]
```diff
@@ -17,6 +17,12 @@ object TestAssetHelper {
     @Suppress("MagicNumber")
     val waitingTime: Long = TimeUnit.SECONDS.toMillis(15)
     val waitingTimeShort: Long = TimeUnit.SECONDS.toMillis(1)
+    // File name length in most operating systems is limited to 255 bytes,
+    // including file extension.
+    // This string is 251 char long though, as AS won't compile with a resource
```
Comment (grigoryk) [inline app/src/androidTest/java/org/mozilla/fenix/helpers/TestAssetHelper.kt:22]: nit: expand AS to Android Studio. AS more often means "application-services" in our context :)

**Review by Amejia481 — COMMENTED** [review]
This looks great!
Congratulation is your first PR 🎉 

There are a few things that we have to take a look before landing :)

* There are some nits from the [ktlint task](https://github.com/mozilla-mobile/fenix/runs/4108477538?check_suite_focus=true), you can run it locally by using `./gradlew ktlint`, additionally it's a good idea to also run `./gradlew detekt`, if you want to auto-correct any `ktlint` format issues you could run `./gradlew  ktlintFormat`.
* There's an [UI test failing](https://console.firebase.google.com/u/1/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8302961783685630124/executions/bs.d6f1791673cac7f8/testcases/1/test-cases) looks like it could be linked to this change  `org.mozilla.fenix.ui.SmokeTest.manageDownloadsInDownloadsMenuTest`

```
 androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: with text: is "l38ID7Xze57vCac8C9oD7Z2LXzBw00HDiw7XR6ZDu5G5O8uyODAVEOTS2PrZt8OoBM77CmaFyrdGxUODuEWwpfzwnTsTTRcGDsr6Cez4Q7DK0Kr0KJIRVAFbV4czwMeiI25FIml6QCMvQR8nBZHe1oUPQn23BplLC4c3iXGvuEBGEhyU81UpqqTSwU5tfxZ7mBOYcQUqYNG0A7ixekg9awVeq8PncVdCZKLA0hXgJEW4.svg"
View Hierarchy:
+>DecorView{id=-1, visibility=VISIBLE, width=1080, height=1920, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params={(0,0)(fillxfill) sim={adjust=resize} ty=BASE_APPLICATION wanim=0x7f14034b
fl=LAYOUT_IN_SCREEN LAYOUT_INSET_DECOR HARDWARE_ACCELERATED DRAWS_SYSTEM_BAR_BACKGROUNDS
pfl=FORCE_DRAW_STATUS_BAR_BACKGROUND
vsysui=LIGHT_STATUS_BAR LIGHT_NAVIGATION_BAR}, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2} 
```


Apart from that this look pretty good, congrats!

**Review by grigoryk — APPROVED** [review]
Paired up on this a bit to go through the linter issues, and feedback from Arturo. Tests, updated UI and code changes all look good to me.

**Comment by Amejia481** [comment]
Thanks @grigoryk ❤️ !

**Comment by sflorean** [comment]
This was verified as fixed, and the bug was updated and closed. I will remove the qa label.

**Pull request merged**